### PR TITLE
Insert picture skill

### PR DESCRIPTION
**Pull request by RaviCompro**

### CODE REVIEW

File: server/libs/skills/common/baseSkill.js [hunk 1]
```diff
@@ -1,64 +1,132 @@
...
+                    return Promise.reject(error);
+                });
+        }
+        else {
+            var resolveParams = { "attrValue": "" };
             return Promise.resolve(resolveParams);
-          }, function (error) {
+        }
+    }
+
```
Comment (RaviCompro) [inline server/libs/skills/common/baseSkill.js:81]: detailed comments

File: server/libs/skills/common/baseSkill.js [hunk 2]
```diff
@@ -1,64 +1,132 @@
...
+    resourcePathWithUpdatedReferences(skillParams) {
+        try {
+            let { resourcePath, docImages } = skillParams.paramsObj;
```
Comment (RaviCompro) [inline server/libs/skills/common/baseSkill.js]: name docImages needs to be changed...

File: server/libs/skills/common/baseSkill.js [hunk 3]
```diff
@@ -1,64 +1,132 @@
...
+            let absolutePath = config.fileStore.resourceFolder + resourcePath;
```
Comment (RaviCompro) [inline server/libs/skills/common/baseSkill.js]: name of var absolutePath

File: server/libs/skills/common/baseSkill.js [hunk 4]
```diff
@@ -1,64 +1,132 @@
...
+            let fromArray = [];
+            let toArray = [];
+            let modifiedPathArr = [];
+
+            docImages.forEach(function (imgObject) {
+                modifiedPathArr.push(skillParams.taskParams.addResourceToMap("step", imgObject.path).absFilePath);
```
Comment (RaviCompro) [inline server/libs/skills/common/baseSkill.js]: fn call needs to be changed

File: server/libs/skills/common/baseSkill.js [hunk 5]
```diff
@@ -1,64 +1,132 @@
...
+            });
+            return new Promise((resolve, reject) => {
+                fs.readFile(absolutePath, 'utf8', (error, htmlResource) => {
```
Comment (RaviCompro) [inline server/libs/skills/common/baseSkill.js]: htmlResource

Comment (RaviCompro) [inline server/libs/skills/common/baseSkill.js]: use fal for reading file

File: server/libs/skills/common/baseSkill.js [hunk 6]
```diff
@@ -1,64 +1,132 @@
...
+                                    }
+                                });
+                            });
+
+                            for (let index = 0; index < fromArray.length; index++) {
+                                htmlResource = htmlResource.replace(fromArray[index], toArray[index]);
+                            }
+                            let resourcePathArray = resourcePath.split('/');
+                            let fileName = resourcePathArray[resourcePathArray.length - 1];
+                            dbfileStoreManager.saveTaskDynamicResource(skillParams.taskParams, htmlResource, fileName)
```
Comment (RaviCompro) [inline server/libs/skills/common/baseSkill.js]: use reference of dbfileStoreManager from skillParams

File: server/libs/skills/common/baseSkill.js [hunk 7]
```diff
@@ -1,64 +1,132 @@
+
+const fs = require('fs');
```
Comment (RaviCompro) [inline server/libs/skills/common/baseSkill.js]: remove fs, config, dbfileStoreManager

File: server/libs/skills/wd/common/wdSkill.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
+const BaseSkill = require("../../common/baseSkill");
+
+//Wrod based Common Functionality goes here 
+
+const xmlUtil = require("../../../../utils/xmlUtil");
+const dbfileStoreManager = require('../../../../modules/skill/dbFilestoreMgr');
+
+module.exports = class WordSkill extends BaseSkill {
+    createCursorPosition(skillParams) {
```
Comment (RaviCompro) [inline server/libs/skills/wd/common/wdSkill.js:9]: detailed comments
define skillParams

File: server/libs/skills/wd/common/wdSkill.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    createTreeXMLPath(skillParams) {
+        return dbfileStoreManager.copyTaskAssetFile(skillParams.paramsObj.treeXML, skillParams.taskParams)
+            .then((resolveParam) => {
+                return Promise.resolve({ "attrValue": resolveParam.filePath });
+            }).catch((error) => {
+                return Promise.reject(error);
+            });
+    }
+
+    createAddressBarImagePath(skillParams) {
```
Comment (RaviCompro) [inline server/libs/skills/wd/common/wdSkill.js:29]: 1. comments
2. confirm if preload resource is required for this attribute or not

File: server/libs/skills/wd/insertPicture/insertPicture.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        try {
+            let paramValueObj = skillParams.paramsObj;
+            let imgName = paramValueObj["imageFileName"];
+            return Promise.resolve({ attrValue: [imgName, imgName.slice(0, imgName.lastIndexOf('.'))] });
+        } catch (error) {
+            return Promise.reject(error);
+        }
+    }
+
+    htmlFileUpload(inputFile, dependantDropzoneModel) {
```
Comment (RaviCompro) [inline server/libs/skills/wd/insertPicture/insertPicture.js]: front end function
can have some naming convention signifying that it's a front end function
option 1: fn name can have a prefix "fe" => this fn name will change to "feHTMLFileUpload"

**Review by dhirendra-singh81 — APPROVED** [review]

**Pull request merged**